### DISCUSSION TITLE
Untagged deserialization of `Query<Result<T, U>>`

### DISCUSSION THREAD

**Discussion by ghost** [tokio-rs/axum#2735] [Q&A]
### Summary

Currently this works but it's ugly and verbose:
```rust
#[derive(Deserialize)]
#[serde(untagged)]
pub enum AuthResult {
    Ok(AuthorizationResponse),
    Err(ClientError),
}
```

I came up with the following snippet but it doesn't compile since I can't reuse `deserializer`.
```rust
@mikoto:kurosaki.cx|
12:52 AM

#[derive(Deserialize)]
pub struct AuthResult(
    #[serde(deserialize_with = "de_auth_result")] Result<AuthResponse, ClientError>,
);

fn de_auth_result<'de, D>(deserializer: D) -> Result<Result<AuthResponse, ClientError>, D::Error>
where
    D: Deserializer<'de>,
{
    AuthResponse::deserialize(deserializer)
        .map(Result::Ok)
        .or_else(|_| ClientError::deserialize(deserializer).map(Result::Err))
}
```

The problem is that the query might contain an error or successful response and I wanna have access to the error to trace it.

### axum version

0.6.20

**Comment by mladedav** [comment]
I think `Result<Query<YourType>>` should provide you with the deserialization error if there is one. Does that not work for you?
